### DISCUSSION TITLE
Choosing right default batch size for pathcore based on gpu memory configuration

### DISCUSSION THREAD

**Discussion by udasinnayan** [open-edge-platform/anomalib#1251] [Q&A] [answered]
In the `pathcore` configuration file, how to increase the batch size based on the GPU memory usage. Right now in the dataset configuration file `config.yaml` there are two options which are `train_batch_size: 32` and `test_batch_size: 32`. In the `trainer` section to choose batch size `limit_train_batches: 1.0`, `limit_val_batches: 1.0`, `limit_test_batches: 1.0`, and `limit_predict_batches: 1.0`. 

If I want to use custom data format based model patchcore algorithm, by default it can chose batch size 2 for training and testing. But if I would like to increase the testing batch size to 4 what part in the config file needs to be updated to use 100% of test dataset with custom batch size based on the gpu memory usage? I tried to use batch size 4 in config file `dataset` section `test_batch_size: 4` but it always chose 2 as a testing batch size? How do I choose a custom batch size? I am using `lightning_inference.py` for testing the model.

Do I have to change any script if I wanna use only gpu for testing? I mean the value of `precision` or something?

TIA

**Comment by samet-akcay** [comment, marked as answer]
@udasinnayan, apologies for my late reply to this. To change the batch size of validation, test and predict dataloaders, you need to configure `eval_batch_size` parameter in the config file.
https://github.com/openvinotoolkit/anomalib/blob/cec86bfb1174c7c1cedd39e8b900384e41553b41/src/anomalib/models/patchcore/config.yaml#L8

Regarding the `limit_<split>_batches` params in `Trainer`, they are more for debugging purposes, which subsamples the corresponding dataloader. For more information, you could refer to this link:
https://lightning.ai/docs/pytorch/stable/common/trainer.html#limit-train-batches

**Comment by samet-akcay** [comment]
Let us know if you have any other follow-up questions. Thanks!
